kafka-ui-react-app/src/components/Topics/Topic/Overview/ActionsCell.tsx

```diff
@@ -7,19 +7,22 @@ import { RouteParamsClusterTopic } from 'lib/paths';
 import useAppParams from 'lib/hooks/useAppParams';
 import { clearTopicMessages } from 'redux/reducers/topicMessages/topicMessagesSlice';
 import { Dropdown, DropdownItem } from 'components/common/Dropdown';
-import { useTopicDetails } from 'lib/hooks/api/topics';
+import { topicKeys, useTopicDetails } from 'lib/hooks/api/topics';
+import { useQueryClient } from '@tanstack/react-query';
 
 const ActionsCell: React.FC<CellContext<Partition, unknown>> = ({ row }) => {
   const { clusterName, topicName } = useAppParams<RouteParamsClusterTopic>();
   const { data } = useTopicDetails({ clusterName, topicName });
   const { isReadOnly } = React.useContext(ClusterContext);
   const { partition } = row.original;
   const dispatch = useAppDispatch();
+  const queryClient = useQueryClient();
 
   const clearTopicMessagesHandler = async () => {
```

Comment:
i think it is best to transform this `redux` functionality clear -> `react-query` , hence the `query` will handle delete as well, this way it is mixed.

Reply:
it is still un-removed here , we should change it like the upper one.

Reply:
i done it